Setting CICE dt directly has no impact (#1501)
COSIMA/MOM6-CICE6: documentation/docs/pages/infrastructure/NUOPC-driver.md

### CICE6 time-steps

> The CICE thermodynamics time-step (`dt`) is set in the [CICE NUOPC cap](https://github.com/ESCOMP/CICE/blob/4cb296c4003014fe57d6d00f86868a78a532fc95/cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90#L1227) to match the driver time-step, which [equals the coupling time-step](#coupling-and-driver-time-step). Setting `dt` in `ice_in` has no impact, although [other time-step related parameters](https://cice-consortium-cice.readthedocs.io/en/latest/user_guide/ug_implementation.html#choosing-an-appropriate-time-step) can be set here. Setting `ndtd` within `ice_in` allows for sub-cycling of the sea-ice dynamics to ensure numerical stability and may need to be increased during initial model spin up (the thermodynamics should be numerically stable for any time-step).

Reviewer: ```suggestion
The CICE thermodynamics time-step (`dt`; seconds) is set in the [CICE NUOPC cap](https://github.com/ESCOMP/CICE/blob/4cb296c4003014fe57d6d00f86868a78a532fc95/cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90#L1227) to match the driver time-step, which [equals the coupling time-step](#coupling-and-driver-time-step). Setting `dt` in `ice_in` has no impact, although [other time-step related parameters](https://cice-consortium-cice.readthedocs.io/en/latest/user_guide/ug_implementation.html#choosing-an-appropriate-time-step) can be set here. Setting `ndtd` within `ice_in` allows for sub-cycling of the sea-ice dynamics to ensure numerical stability and may need to be increased during initial model spin up (the thermodynamics should be numerically stable for any time-step).
```

> Setting `dt` in `ice_in` has no impact, although [other time-step related parameters](https://cice-consortium-cice.readthedocs.io/en/latest/user_guide/ug_implementation.html#choosing-an-appropriate-time-step) can be set here. 

So this is a result of the use of the nuopc cap?

In some ways I prefer the old wording as it tells the user why. [Looking at the docs](https://cice-consortium-cice.readthedocs.io/en/latest/user_guide/ug_implementation.html#choosing-an-appropriate-time-step) that are linked makes one think it should be important. I think adding a few words on why it doesn't matter will reduce confusion.

> The CICE thermodynamics time-step (`dt`) is set in the [CICE NUOPC cap](https://github.com/ESCOMP/CICE/blob/4cb296c4003014fe57d6d00f86868a78a532fc95/cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90#L1227) to match the driver time-step, which [equals the coupling time-step](#coupling-and-driver-time-step). Setting `dt` in `ice_in` has no impact, although [other time-step related parameters](https://cice-consortium-cice.readthedocs.io/en/cice6.5.0/user_guide/ug_implementation.html#choosing-an-appropriate-time-step) can be set here. Setting `ndtd` within `ice_in` allows for sub-cycling of the sea-ice dynamics to ensure numerical stability and may need to be increased during initial model spin up (the thermodynamics should be numerically stable for any time-step).

Author: ```suggestion
The CICE thermodynamics time-step (`dt`) is set in the [CICE NUOPC cap](https://github.com/ESCOMP/CICE/blob/4cb296c4003014fe57d6d00f86868a78a532fc95/cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90#L1227) to match the driver time-step, which [equals the coupling time-step](#coupling-and-driver-time-step). Setting `dt` in `ice_in` has no impact, although [other time-step related parameters](https://cice-consortium-cice.readthedocs.io/en/latest/user_guide/ug_implementation.html#choosing-an-appropriate-time-step) can be set here. Setting `ndtd` within `ice_in` allows for sub-cycling of the sea-ice dynamics to ensure numerical stability and may need to be increased during initial model spin up (the thermodynamics should be numerically stable for any time-step).
```

> The CICE thermodynamics time-step (`dt`; seconds) is set in the [CICE NUOPC cap](https://github.com/ESCOMP/CICE/blob/4cb296c4003014fe57d6d00f86868a78a532fc95/cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90#L1227) to match the driver time-step, which [equals the coupling time-step](#coupling-and-driver-time-step). Setting `dt` in `ice_in` has no impact, although [other time-step related parameters](https://cice-consortium-cice.readthedocs.io/en/latest/user_guide/ug_implementation.html#choosing-an-appropriate-time-step) can be set here. Setting `ndtd` within `ice_in` allows for sub-cycling of the sea-ice dynamics to ensure numerical stability and may need to be increased during initial model spin up (the thermodynamics should be numerically stable for any time-step).

Author: ```suggestion
The CICE thermodynamics time-step (`dt`; seconds) is set in the [CICE NUOPC cap](https://github.com/ESCOMP/CICE/blob/4cb296c4003014fe57d6d00f86868a78a532fc95/cicecore/drivers/nuopc/cmeps/ice_comp_nuopc.F90#L1227) to match the driver time-step, which [equals the coupling time-step](#coupling-and-driver-time-step). 
Setting `dt` in `ice_in` has no impact and can only be changed by changing the coupling timestep. [Other time-step related parameters (`ndtd` and `ndte`)](https://cice-consortium-cice.readthedocs.io/en/latest/user_guide/ug_implementation.html#choosing-an-appropriate-time-step) can be set in `ice_in`. Setting `ndtd` within `ice_in` allows for sub-cycling of the sea-ice dynamics to ensure numerical stability and may need to be increased during initial model spin up (the thermodynamics should be numerically stable for any time-step).
```

Author: How this suggestion sound @chrisb13 ?

Reviewer: Ta, looks good.

- Baroclinic time-step (`DT`) for Lagrangian stacked shallow-water equations; often called "the" model timestep; needs to be short enough to resolve internal gravity waves, inertial oscillations and advection on the horizontal grid (i.e. this is resolution-dependent).

> - Tracer/thermodynamics time-step (`DT_THERM`), which can be [set to resolve the relevant physics](https://youtu.be/JKMwd8VXYcU?si=dBk09lMVPsMF3aJT&t=541) (e.g. an hour or so to capture the diurnal cycle), independent of the horizontal grid resolution. It is possible to set `DT_THERM` longer than the coupling time-step, but not with [`DIABATIC_FIRST = True`](https://github.com/search?q=repo%3AACCESS-NRI%2Faccess-om3-configs+path%3Adoc%2FMOM_input+content%3ADIABATIC_FIRST&type=code), which is the case for the current ACCESS-OM3 configurations. So `DT_THERM` should be set equal to, or less than, the coupling time-step.

Reviewer: >  (e.g. an hour or so to capture the diurnal cycle)

Currently 1 or 2 hours, see

https://github.com/ACCESS-NRI/access-om3-configs/pull/1504/changes#r3575910974

Reviewer: > So `DT_THERM` should be set equal to, or less than, the coupling time-step.

How is this a "So"? (Noting that this sentence was in the previous version.)

Author: Not in scope, I think using the web interface added or deleted a new line at the end.

Reviewer: I'm not sure if you've understood my comment? 

With this sentence:
> So `DT_THERM` should be set equal to, or less than, the coupling time-step.

I don't understand how the "so" is following from the previous sentence, do you?

If we both don't then I propose that we delete it. As in:
> `DT_THERM` should be set equal to, or less than, the coupling time-step.

I realise this isn't one of your proposed changes but this seems trivial enough to fix now doesn't it?

Author: See https://github.com/ACCESS-NRI/access-om3-configs/pull/1504
